src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiSchemaGenerator.cs

```diff
@@ -473,6 +474,16 @@ internal static OpenApiSchema CreateSchemaTypeSchema(this ODataContext context,
                     AnyOf = null
                 };
 
+                if (schema.Discriminator != null)
+                {
+                    schema.Properties.TryAdd(discriminatorPropertyName, new OpenApiSchema()
+                    {
+                        Type = "string"                        
+                    });
+                    schema.Properties[discriminatorPropertyName].Default = new OpenApiString("#" + structuredType.FullTypeName());
```

Comment:
```suggestion
                        Type = "string",
                       .Default = new OpenApiString("#" + structuredType.FullTypeName()),
                    });
```

Reply:
No. I intentionally set the `Default` outside the `TryAdd()` method so as to capture for all cases, including those where the `discriminatorPropertyName` -> `@odata.type` property already exists in the schema.

Reply:
in what scenario could the property could have already been added?

Reply:
In your [comment](https://github.com/microsoft/OpenAPI.NET.OData/issues/243#issuecomment-1168726146) in the issue you mentioned that some base types are missing this property and that we should walk the inheritance tree to make sure the property is present and add it if not.

Reply:
Ah my bad, I should probably have added another comment with the latest requirement to clear any confusion.
The current implementation adds the property to all entity or complex types (which is what we now want). TryAdd would never have checked the presence of the property in a base schema, for that we'd have had to walk the allOf properties recursively. But since we don't want to add the property only on the base type anymore this requirement is moot.
I can't foresee a case where the odata type property would already be here besides it being described in the CSDL, and using convention reserved property names would be horribly wrong, and the conversion should fail.

Bottom line, move the default above, and if the tryadd returns false, throw an invalid operation exception saying the property is already present, which is not expected, and the CSDL should be verified.

I hope this makes things clearer, don't hesitate if you have follow up questions.